Modules/_functoolsmodule.c

```diff
@@ -399,8 +401,28 @@ partial_repr(partialobject *pto)
         if (arglist == NULL)
             goto done;
     }
-    result = PyUnicode_FromFormat("%s(%R%U)", Py_TYPE(pto)->tp_name,
-                                  pto->fn, arglist);
+
+    mod = _PyType_GetModuleName(Py_TYPE(pto));
+
+    if (mod == NULL) {
+        name = PyType_GetQualName(Py_TYPE(pto));
+        if (name == NULL) {
+            return NULL;
+        }
+        result = PyUnicode_FromFormat("%S(%R%U)", name, pto->fn, arglist);
```

Comment:
You should just return NULL here, unless you have a good reason to handle the partial type without `__module__` (but then the code should be more complicated)